components/engine_traits/src/range_cache_engine.rs

```diff
@@ -105,37 +103,58 @@ pub trait RangeCacheEngineExt {
 /// as it continues to evolve to handle eviction, using stats.
 pub trait RangeHintService: Send + Sync {}
 
-#[derive(Clone, Eq, PartialEq, PartialOrd, Ord)]
-pub struct CacheRange {
+#[derive(Clone, Eq, PartialEq)]
+pub struct CacheRegion {
+    // target region id
+    pub id: u64,
+    // the version of target region epoch. we only track version but not
+    // conf_version because conf_version does not change the applied data.
+    pub epoch_version: u64,
+    // data start key of the region range,  equals to data_key(region.start_key).
     pub start: Vec<u8>,
+    // data start key of the region range, equals to data_end_key(region.start_key).
     pub end: Vec<u8>,
 }
 
-impl Debug for CacheRange {
+impl Debug for CacheRegion {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("CacheRange")
+        f.debug_struct("CacheRegion")
+            .field("id", &self.id)
+            .field("epoch", &self.epoch_version)
             .field("range_start", &log_wrappers::Value(&self.start))
             .field("range_end", &log_wrappers::Value(&self.end))
             .finish()
     }
 }
 
-impl CacheRange {
-    pub fn new(start: Vec<u8>, end: Vec<u8>) -> Self {
-        Self { start, end }
+impl CacheRegion {
+    pub fn new<T1: Into<Vec<u8>>, T2: Into<Vec<u8>>>(
+        id: u64,
+        epoch_version: u64,
+        start: T1,
+        end: T2,
+    ) -> Self {
+        Self {
+            id,
+            epoch_version,
+            start: start.into(),
+            end: end.into(),
```

Comment:
How about adding asserts `start` and `end` that they start with 'z'?

Reply:
`CacheRange::new` is only used for test, seems some old cases uses ranges that are not actually fall in the data range.